src/jsdoc_transformer.ts

```diff
@@ -542,6 +542,24 @@ export function jsdocTransformer(
         return memberDecl ? [decl, memberDecl] : [decl];
       }
 
+      /** Returns the `this` type in this context, or undefined if none. */
+      function getContextThisType(node: ts.Node): ts.Type|undefined {
+        do {
+          if (ts.isClassDeclaration(node) && node.name) {
+            return typeChecker.getTypeAtLocation(node.name);
+          }
+          if (ts.isFunctionDeclaration(node) && node.parameters.length > 0) {
+            const firstParam = node.parameters[0];
+            if (ts.isIdentifier(firstParam.name) &&
+                transformerUtil.getIdentifierText(firstParam.name) === 'this' && firstParam.type) {
+              return typeChecker.getTypeAtLocation(firstParam.type);
```

Comment:
This feels a bit sketchy to me, I wonder if it's better to ask for the function's type and get it from there?

Reply:
The TS API does not expose this type declarations on signature, so we cannot do much better here. Left a comment.